crates/bin/prove_block/src/main.rs

```diff
@@ -290,8 +340,24 @@ async fn main() -> Result<(), Box<dyn Error>> {
         }
     }
 
+    // TODO: clean up...
+    // now that we have the class_hash for each contract, fill in the transaction data with the
+    // class_hash. when transactions were first processed above, this information wasn't available.
+    for transaction in transactions.iter_mut() {
+        if let Some(sender_address) = transaction.sender_address {
+            let class_hash = address_to_class_hash
+                .get(&sender_address)
+                .expect("should have a class_hash for each known contract addresses at this point");
+            log::info!("Filling in class_hash {:x} for txn", class_hash);
+            transaction.class_hash = Some(*class_hash);
+        } else {
+            // TODO: are there txn types which wouldn't have a sender address?
+            log::warn!("Found transaction without sender_address");
```

Comment:
Did you encounter some of these?
maybe its better to just put an unimplemented?

Reply:
I didn't encounter any, but it's an `Option` so I'm guessing it's possible for some older txn versions or something like that.

Reply:
Converted to `unimplemented!()` in https://github.com/Moonsong-Labs/snos/pull/19/commits/b514f57365087bb88a617b237778c0a9d20a28f0